src/platform/silabs/efr32/BLEChannel.h

```diff
@@ -0,0 +1,188 @@
+#pragma once
+
+#include "sl_bt_api.h"
+#include "sl_bt_rtos_adaptation.h"
+#include "sl_bt_stack_config.h"
+#include "sl_bt_stack_init.h"
+#include <lib/core/CHIPError.h>
+#include <lib/core/Optional.h>
+#include <lib/support/BitFlags.h>
+#include <lib/support/Span.h>
+
+namespace chip {
+namespace DeviceLayer {
+namespace Internal {
+
+struct BLEConState
+{
+    uint16_t mtu : 10;
+    uint16_t allocated : 1;
+    uint16_t subscribed : 1;
+    uint16_t unused : 4;
+    uint8_t connectionHandle;
+    uint8_t bondingHandle;
+};
+
+struct AdvConfigStruct
+{
+    ByteSpan advData;
+    ByteSpan responseData;
+    uint32_t intervalMin;
+    uint32_t intervalMax;
+    uint16_t duration;
+    uint8_t maxEvents;
+};
+
+class BLEChannel
+{
+public:
+    BLEChannel()
+    {
+        // Initialize the connection state
+        memset(&mConnectionState, 0, sizeof(mConnectionState));
+        mFlags.ClearAll();
+        mAdvIntervalMin = 0;
+        mAdvIntervalMax = 0;
+        mAdvDuration    = 0;
+        mAdvMaxEvents   = 0;
+    }
+
+    virtual ~BLEChannel() = default;
+
+    virtual CHIP_ERROR Init(void) = 0;
+
+    // TODO: Revisit this method and see about splitting timing vs advertising data. It is currently implemented to do everything
+    /** @brief ConfigureAdvertising
+     *   Configure the advertising data and parameters for the BLE channel. This function needs to be called before
+     *  starting the advertising process.
+     *
+     *  @param AdvConfigStruct Structure holding The advertising data to be set, the scan response data to be set,
+     *  the minimum and maximum advertising intervals, the duration of the advertising, and the maximum number of events.
+     */
+    virtual CHIP_ERROR ConfigureAdvertising(const AdvConfigStruct & config) = 0;
+
+    /** @brief StartAdvertising
+     *  Start the advertising process for the BLE channel using configured parameters. ConfigureAdvertising must be called before
+     * this function.
+     */
+    virtual CHIP_ERROR StartAdvertising(void) = 0;
+    virtual CHIP_ERROR StopAdvertising(void)  = 0;
+
+    virtual void AddConnection(uint8_t connectionHandle, uint8_t bondingHandle)       = 0;
+    virtual bool RemoveConnection(uint8_t connectionHandle)                           = 0;
+    virtual void HandleReadRequest(volatile sl_bt_msg_t * evt, ByteSpan data)         = 0;
+    virtual void HandleWriteRequest(volatile sl_bt_msg_t * evt, MutableByteSpan data) = 0;
+
+    /** @brief CanHandleEvent
+     *  Determine if the side channel can handle a BLE event. This only gets called if the
+     *  targetted event is not supported by MATTERoBLE as the side channel supports every
+     *  event that MATTERoBLE does.
+     *
+     * @param event uint32_t Header of sl_bt_msg_t event type
+     * @return true if the event can be handled, false otherwise.
+     */
+    virtual bool CanHandleEvent(uint32_t event) { return false; }
+
+    virtual void ParseEvent(volatile sl_bt_msg_t * evt) {}
```

Comment:
Why are these the only functions that are not pure virtual?

Reply:
Because the ability to parse events in the Channel will be implemented in the future.

Reply:
I would add a comment with a TODO then so that we remember this info